docs/user-guide/managing-deployments.md

```diff
@@ -499,20 +499,34 @@ A more disciplined alternative to patch and edit is `kubectl apply`.
 
-This command will compare the version of the configuration that you're pushing with the previous version and apply the changes you've made, without overwriting any automated changes to properties you haven't specified.
+This command will compare the version of the configuration that you're pushing with the previous version and apply the changes you've made, without overwriting any changes to properties you haven't specified.
 
 ```console
 $ kubectl apply -f ./nginx-rc.yaml
 replicationcontroller "my-nginx-v4" configured
 ```
 
-As shown in the example above, the configuration used with `kubectl apply` is the same as the one used with `kubectl replace`. However, instead of deleting the existing resource and replacing it with a new one, `kubectl apply` modifies the configuration of the existing resource.
+As shown in the example above, the configuration used with `kubectl apply` is the same as the one used with `kubectl replace`.
+However, instead of deleting the existing resource and replacing it with a new one, `kubectl apply` modifies the configuration of the existing resource.
 
-Note that `kubectl apply` attaches an annotation to the resource in order to determine the changes to the configuration since the previous invocation. When it's invoked, `kubectl apply` does a three-way diff between the previous configuration, the provided input and the current configuration of the resource, in order to determine how to modify the resource.
+Note that `kubectl apply` uses an annotation, which stores the previous configuration of the resource supplied to `kubectl`.
+When it's invoked, `kubectl apply` does a three-way diff between the previous configuration, the configuration supplied to the current invocation of `kubectl`,
+and the current configuration of the resource returned by the api server.
 
-Currently, resources are created without this annotation, so the first invocation of `kubectl apply` will fall back to a two-way diff between the provided input and the current configuration of the resource. During this first invocation, it cannot detect the deletion of properties set when the resource was created. For this reason, it will not remove them.
+Properties present in the previous configuration that are not present in the supplied configuration are removed from the current configuration of the resource,
+and properties added or changed in the supplied configuration are added to changed in the current configuration of the resource, respectively.
 
-All subsequent calls to `kubectl apply`, and other commands that modify the configuration, such as `kubectl replace` and `kubectl edit`, will update the annotation, allowing subsequent calls to `kubectl apply` to detect and perform deletions using a three-way diff.
+By default, resources are created without the annotation.
+You can override the default and cause the annotation to be applied by `kubectl` commands that modify the configuration using the `--save-config` flag.
```

Comment:
not all kubectl commands support --save-config (currently only edit, replace, expose, run, autoscale, and create supports that)